presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/BaseCostBasedPlanTest.java

```diff
@@ -0,0 +1,245 @@
+import static java.nio.file.Files.exists;
+import static java.util.Locale.ENGLISH;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.joining;
+import static org.testng.Assert.assertEquals;
+
+public abstract class BaseCostBasedPlanTest
+        extends BasePlanTest
+{
+    private final Path mavenModulePath;
+
+    public BaseCostBasedPlanTest(String mavenModule, LocalQueryRunnerSupplier supplier)
+    {
+        super(supplier);
+        this.mavenModulePath = Paths.get(requireNonNull(mavenModule, "mavenModulePath is null"));
+    }
+
+    protected abstract Stream<Query> getQueries();
+
+    @DataProvider
+    public Object[][] getQueriesDataProvider()
+    {
+        return getQueries()
+                .collect(toDataProvider());
+    }
+
+    @Test(dataProvider = "getQueriesDataProvider")
+    public void test(Query query)
+    {
+        String actual = joinOrderString(resolve(query.query))
+                .stream()
+                .collect(joining("\n"));
+        assertEquals(actual, read(resolve(getExpectedJoinOrderingFile(query.id))));
+    }
+
+    private Path resolve(Path path)
+    {
+        if (exists(path)) {
+            return path;
+        }
+        Path fromParent = Paths.get("..").resolve(path);
+        checkState(exists(fromParent), "Unable to resolve: " + path);
+        return fromParent;
+    }
+
+    public void generate()
+            throws Exception
+    {
+        initPlanTest();
+        try {
+            getQueries()
+                    .parallel()
+                    .forEach(query -> {
+                        try {
+                            Files.write(getExpectedJoinOrderingFile(query.id), joinOrderString(query.query), UTF_8);
+                        }
+                        catch (IOException e) {
+                            throw new RuntimeException(e);
+                        }
+                    });
+        }
+        finally {
+            destroyPlanTest();
+        }
+    }
+
+    protected static String read(Path file)
+    {
+        try {
+            return Files.lines(file).collect(joining("\n"));
+        }
+        catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private Path getExpectedJoinOrderingFile(String queryId)
+    {
+        return mavenModulePath.resolve(Paths.get("src", "test", "resources", mavenModulePath.getFileName().toString(), "join_ordering", queryId + ".txt"));
+    }
+
+    public static final class Query
+    {
+        private final String id;
+        private final Path query;
+
+        public Query(String id, Path query)
+        {
+            this.id = requireNonNull(id, "id is null");
+            this.query = requireNonNull(query, "query is null");
+        }
+
+        @Override
+        public String toString()
+        {
+            return id;
+        }
+    }
+
+    protected List<String> joinOrderString(Path sqlPath)
```

Comment:
The name of this method is confusion. How about `generateQueryPlan`, and change the parameter name `queryFilePath`